ts/interactions/conversations/unsendingInteractions.ts

```diff
@@ -365,27 +377,36 @@ export async function deleteMessagesById(messageIds: Array<string>, conversation
     await Promise.all(messageIds.map(m => getMessageById(m, false)))
   );
 
+  const messageCount = selectedMessages.length;
   const moreThanOne = selectedMessages.length > 1;
 
-  window.inboxStore?.dispatch(
-    updateConfirmModal({
-      title: window.i18n('deleteJustForMe'),
-      message: moreThanOne
-        ? window.i18n('deleteMessagesQuestion')
-        : window.i18n('deleteMessageQuestion'),
-      okText: window.i18n('delete'),
-      okTheme: SessionButtonColor.Danger,
-      onClickOk: async () => {
-        await doDeleteSelectedMessages({
-          selectedMessages,
-          conversation,
-          deleteForEveryone: false,
-        });
-        window.inboxStore?.dispatch(updateConfirmModal(null));
-      },
-      closeAfterInput: false,
-    })
-  );
+  if (window.getSettingValue('confirm-deletions')) {
+    window.inboxStore?.dispatch(
+      updateConfirmModal({
+        title: window.i18n('deleteJustForMe'),
+        message: moreThanOne
+          ? window.i18n('deleteMessagesQuestion', [messageCount.toString()])
+          : window.i18n('deleteMessageQuestion'),
+        okText: window.i18n('delete'),
+        okTheme: SessionButtonColor.Danger,
+        onClickOk: async () => {
+          await doDeleteSelectedMessages({
+            selectedMessages,
+            conversation,
+            deleteForEveryone: false,
+          });
+          window.inboxStore?.dispatch(updateConfirmModal(null));
+        },
+        closeAfterInput: false,
+      })
+    );
+  } else {
+    await doDeleteSelectedMessages({
+      selectedMessages,
+      conversation,
+      deleteForEveryone: false
+    });
+  }
```

Comment:
same comment as above, create a temporary function to avoid duplication

Reply:
Changes made as requested. Rebased on current `clearnet`.